libtransmission/torrent.cc

```diff
@@ -2453,6 +2453,37 @@ bool renameArgsAreValid(std::string_view oldpath, std::string_view newname)
         !tr_strvContains(newname, TR_PATH_DELIMITER);
 }
 
+bool renameNoCollision(tr_torrent const* tor, std::string_view oldpath, std::string_view newname)
+{
+    std::string newpath{};
+    if (tr_strvContains(oldpath, TR_PATH_DELIMITER))
+    {
+        newpath = fmt::format(FMT_STRING("{:s}/{:s}"sv), tr_sys_path_dirname(oldpath), newname);
+    }
+    else
+    {
+        newpath = newname;
+    }
+    if (newpath == oldpath)
+    {
+        return true;
+    }
+
+    auto const newpath_as_dir = tr_pathbuf{ newpath, '/' };
+    auto const n_files = tor->file_count();
+
+    for (tr_file_index_t i = 0; i < n_files; ++i)
+    {
+        auto const& name = tor->file_subpath(i);
+        if (name == newpath || tr_strvStartsWith(name, newpath_as_dir))
```

Comment:
If `newpath` is a `tr_pathbuf` as per the previous comment, I think the `==` will need to be reversed to make the compiler happy, e.g.

```suggestion
        if (newpath == name || tr_strvStartsWith(name, newpath_as_dir))
```